[debops.cryptsetup] Support an macOS Ansible controller. Fixes #31.

ypid opened the pull request:
It has been checked that `head --byte` behaves like `iflag=fullblock` and the role has been locally tested with this change.

Fixes: #31 

I don’t think this needs to be merged right away. I would rather have some times and ideally multiple reviewers. Mandatory is one DebOps Developer and one guy using macOS.

Note that support for macOS in DebOps is provided on best effort bases. No DebOps Developer uses macOS.

@jinnko Can you try that this works on macOS?

ypid committed:
[debops.cryptsetup] Support an macOS Ansible controller. Fixes #31.

It has been checked that `head --byte` behaves like `iflag=fullblock`.

> "accumulate full blocks of input" is of crucial importance here as explained in [Create random data with dd and get “partial read warning”. Is the data after the warning now really random?](https://unix.stackexchange.com/questions/121865/create-random-data-with-dd-and-get-partial-read-warning-is-the-data-after-the). Without the `iflag=fullblock` and because of the blocking nature of `/dev/random` it would be very likely that `dd` only reads partial blocks which will result in a much smaller key size than what we specified. In the worst case, dd would only return one byte or maybe even none at all which would then be used as your key. I guess I don’t need to tell you how easy that would be to brute force. We can not allow this to happen.

drybjed reviewed (dismissed):
Looks good to me, but I'll wait with merging for other reviewers who use macOS to confirm the change works as expected.

stek29 commented:
There's no `--byte` option on BSD/Apple `head`.
`head -c X` is the right command for it -- see [manpage](http://www.manpagez.com/man/1/head/)

Also, from what I see even GNU head doesn't have "--byte" option advised in man/help (however, it works, so I guess it's just left for backward compatibility)

Both GNU and BSD head support `-c` option, and it is in fact equivalent to `--byte` afaiu:

```
❯ ghead --help|grep -- --byte
  -c, --bytes=[-]NUM       print the first NUM bytes of each file;
```

And I assume that `--byte` is same as `-c`:

```
❯ ghead --byte
ghead: option '--bytes' requires an argument
Try 'ghead --help' for more information.
```

ypid committed:
[debops.cryptsetup] There is no `head --byte` or `head --bytes` on macOS

https://github.com/debops/debops/pull/232#issuecomment-374000556

There's no `--byte` option on BSD/Apple `head`.
`head -c X` is the right command for it -- see [manpage](http://www.manpagez.com/man/1/head/)

Also, from what I see even GNU head doesn't have "--byte" option advised in man/help (however, it works, so I guess it's just left for backward compatibility)

Both GNU and BSD head support `-c` option, and it is in fact equivalent to `--byte` afaiu:

```
❯ ghead --help|grep -- --byte
  -c, --bytes=[-]NUM       print the first NUM bytes of each file;
```

And I assume that `--byte` is same as `-c`:

```
❯ ghead --byte
ghead: option '--bytes' requires an argument
Try 'ghead --help' for more information.
```

Thanks to: @stek29

ypid commented:
@stek29 Thanks. `--byte` seems to work because it is unambiguous. This works with most (all?) GNU coreutils. Regardless, `--bytes` should have been used because `--byte` is not documented. I changed it to `-c`.

drybjed reviewed (approved):
OK guys, let me know if the proposed change works on macOS. :-)
